exonum-java-binding/core/rust/src/proxy/runtime.rs

```diff
@@ -134,20 +134,28 @@ impl Runtime for JavaRuntimeProxy {
     fn initiate_adding_service(
         &self,
         context: ExecutionContext<'_>,
-        spec: &InstanceSpec,
+        artifact_id: &ArtifactId,
         parameters: Vec<u8>,
     ) -> Result<(), ExecutionError> {
+        // TODO: change Java API to accept ArtifactId, not InstanceSpec
```

Comment:
Native method lost InstanceSpec parameter, so we constructing it here in order not to break Java code. Must be changed in the future (along with Java changes). initiate_resuming_service is the same

Reply:
Why would we change that if the {name, id} are required to instantiate a service?
         I'd rather have it as an explicit parameter and not buried in the ExecutionContext.

We can consider making an ExecutionContext proxy (when implementing service interfaces);
         or passing {name, id, artifact_id} separately if it makes native implementation
         easier.

Reply:
passing name, id, artifact_id would work, I suppose